### ISSUE TITLE
Certbot on Debian renews certs automatically

### ISSUE THREAD

**Issue by rexbron** [jellyfin/jellyfin-docs#291]
As per title, certbot on Debian is set to automatically renew certificates, no further user interaction required.

Edit documentation to reflect this.

**Comment by castillo92** [comment]
Are you sure?

In my personal notes for setting up a Debian server I have a cron task to execute cerbot and make the checks or update if it is necessary:

`sudo crontab -e`
> # Renew SSL certs if necessary. Everyday @3AM
> 0 3 * * * cd /etc/letsencrypt/ && ./certbot-auto renew && /etc/init.d/apache2 restart

**Comment by rexbron** [comment]
[Certbot Debian Buster (stable) Package](https://packages.debian.org/buster/all/certbot/filelist)

```
/lib/systemd/system/certbot.service
/lib/systemd/system/certbot.timer
```

```
$ systemctl status certbot.timer
● certbot.timer - Run certbot twice daily
   Loaded: loaded (/lib/systemd/system/certbot.timer; enabled; vendor preset: enabled)
   Active: active (waiting) since Wed 2020-04-01 14:04:44 EDT; 2 weeks 1 days ago
  Trigger: Thu 2020-04-16 17:23:39 EDT; 58min left
```

**Comment by castillo92** [comment]
> [Certbot Debian Buster (stable) Package](https://packages.debian.org/buster/all/certbot/filelist)
> 
> ```
> /lib/systemd/system/certbot.service
> /lib/systemd/system/certbot.timer
> ```
> 
> ```
> $ systemctl status certbot.timer
> ● certbot.timer - Run certbot twice daily
>    Loaded: loaded (/lib/systemd/system/certbot.timer; enabled; vendor preset: enabled)
>    Active: active (waiting) since Wed 2020-04-01 14:04:44 EDT; 2 weeks 1 days ago
>   Trigger: Thu 2020-04-16 17:23:39 EDT; 58min left
> ```

I see, thats because you use the package of the official repo.

I used `sudo wget https://dl.eff.org/certbot-auto` and it do not set up those timers. Those servicer are not in my server.

So it depends of the installation

**Comment by rexbron** [comment]
I think it is far more likely that someone would install the package from the debian repo than not.

Any particular reason why you did not?

**Comment by Artiume** [comment]
I often don't use the repo since the packages can be out of date. Maybe we just add a note saying if it's not installed via repo, you must set up renewals manually?

**Comment by castillo92** [comment]
> I think it is far more likely that someone would install the package from the debian repo than not.
> 
> Any particular reason why you did not?

Like @Artiume, I download de package directly from the official website because is the last version. Only that.


EDIT:
I see the package cerbot-buster10 came with automatic setup of cronjob and timers (personally, I do not tried that option, is new to me)

https://certbot.eff.org/lets-encrypt/debianbuster-apache

But the package cerbot-debian-other needs manual setup of cronjob and timers. (I used that option because some years ago, the above package gave me troubles in the installation)

https://certbot.eff.org/lets-encrypt/debianother-apache



So... It depends of the installation that user choose. We could add an advice to check if automatic renew is installed correctly. What do you think?

**Comment by rexbron** [comment]
> Like @Artiume, I download de package directly from the official website because is the last version. Only that.

Not trying to turn this into a flame war but that is at odds with running Debian Stable. Often the dependencies you need need updating as well. If you want to run bleeding edge, Debian Stable is not the distro for you. There are many downstream distros (or Debian Testing) that address software recency. I like the slow release cadence, as I need to set aside at least a day for a dist upgrade to make sure that nothing has broken. 

Secondly, on the matter at hand, I think it best we advise people to install the packages provided by their distro first unless there is a compelling reason not to as a general rule. The distro provides security updates without much additional user interaction.  wgetting your own packages requires much more hands on maintenance.

> https://certbot.eff.org/lets-encrypt/debianother-apache

> So... It depends of the installation that user choose. We could add an advice to check if automatic renew is installed correctly. What do you think?

If you look at other install guides for Debian, Lets Encrypt has been recommending the official Debian packages since Debian 9 (now old-stable, released June 17th, 2017). If you are still running an older version of Debian than old-stable, it shouldn't be internet connected, there are no security updates for the OS anymore! (Debian 8 LTS security isn't done by the normal security team)

So other than a general feeling that Debian doesn't have the latest software, there isn't a good reason to not install from the Debian repos IMO and that should be the first recommended option.

**Comment by Artiume** [comment]
> If you look at other install guides for Debian, Lets Encrypt has been recommending the official Debian packages since Debian 9 (now old-stable, released June 17th, 2017). 

I think the best option is to do this. Recommend the default method, and then add a note or warning that alternative methods may not setup automatic renewals. I'd even be alright to add in the cronjob showing an example of how to manually manage it (why do you renew daily if I may ask?)

**Comment by castillo92** [comment]
> > Like @Artiume, I download de package directly from the official website because is the last version. Only that.
> 
> Not trying to turn this into a flame war but that is at odds with running Debian Stable. Often the dependencies you need need updating as well. If you want to run bleeding edge, Debian Stable is not the distro for you. There are many downstream distros (or Debian Testing) that address software recency. I like the slow release cadence, as I need to set aside at least a day for a dist upgrade to make sure that nothing has broken.
> 
> Secondly, on the matter at hand, I think it best we advise people to install the packages provided by their distro first unless there is a compelling reason not to as a general rule. The distro provides security updates without much additional user interaction. wgetting your own packages requires much more hands on maintenance.
> 
> > https://certbot.eff.org/lets-encrypt/debianother-apache
> 
> > So... It depends of the installation that user choose. We could add an advice to check if automatic renew is installed correctly. What do you think?
> 
> If you look at other install guides for Debian, Lets Encrypt has been recommending the official Debian packages since Debian 9 (now old-stable, released June 17th, 2017). If you are still running an older version of Debian than old-stable, it shouldn't be internet connected, there are no security updates for the OS anymore! (Debian 8 LTS security isn't done by the normal security team)
> 
> So other than a general feeling that Debian doesn't have the latest software, there isn't a good reason to not install from the Debian repos IMO and that should be the first recommended option.

Yes, I am totally agree with you, Debian Stable is like that because it is very stable and safe. I was not pretend to cheer people to not install certbot from official repo.

To not divert more the topic, I think a little note or warning could be great, checking the cronjob waste little time and I think it is a good practise know all possible facts about the software
